### ISSUE TITLE
Issue sending attachments with Unicode filenames

### ISSUE THREAD

**Issue by mwoodbri** [andris9/Nodemailer#496]
Not entirely sure whether it's legitimate to use filenames like this but it's something that Outlook does.

Running this code:
```
require('nodemailer').createTransport({
  host: '...',
  auth: {...}
}).sendMail({
  from: '...',
  to: '...',
  attachments: [{
    filename: '😊.txt'
  }]
});
```
Results in:
```
URIError: URI malformed
    at encodeURIComponent (native)
    at Object.module.exports.buildHeaderParam (/private/tmp/node_modules/libmime/src/libmime.js:512:27)
    at Object.<anonymous> (/private/tmp/node_modules/libmime/src/libmime.js:292:25)
    at Array.forEach (native)
    at Object.module.exports.buildHeaderValue (/private/tmp/node_modules/libmime/src/libmime.js:288:46)
    at /private/tmp/node_modules/buildmail/src/buildmail.js:453:33
    at Array.forEach (native)
    at MimeNode.buildHeaders (/private/tmp/node_modules/buildmail/src/buildmail.js:442:19)
    at MimeNode.stream (/private/tmp/node_modules/buildmail/src/buildmail.js:613:29)
    at MimeNode.createReadStream (/private/tmp/node_modules/buildmail/src/buildmail.js:505:10)
```
Would be very grateful for any help/advice.

p.s. Nodemailer is a great library!

**Comment by andris9** [comment]
Fixed in v1.6.0

**Comment by mwoodbri** [comment]
Fantastic. Thanks!
